sewer/dns_providers/gandi.py

```diff
@@ -0,0 +1,133 @@
+import os
+import requests
+
+from . import common
+
+
+class GandiDns(common.BaseDns):
+    """
+    """
+
+    dns_provider_name = "gandi"
+
+    def __init__(
+        self,
+        GANDI_API_KEY=None,
+        GANDI_API_BASE_URL="https://dns.api.gandi.net/api/v5/",
+        DEFAULT_TTL=10800,
+        requests_lib=requests
```

Comment:
There is an option to pass in a requests library here so that I can pass in a mock requests library. I do this because I was struggling to get python to apply my mock correctly in this file (yes I have googled it a lot). If you have any suggetions I'll happily listen.